2-repositories/scala-repo-server/src/main/scala/com.snowplowanalytics.iglu.server/service/SchemaService.scala

```diff
@@ -245,223 +258,268 @@ class SchemaService(schemaActor: ActorRef, apiKeyActor: ActorRef)
+    * Route to retrieve every schema belonging to a vendor.
+    * @param v list of schema vendors
+    * @param o the owner of the API key the request was made with
+    * @param p API key's permission
+    */
+  @Path("/{vendor}")
+  @ApiOperation(value = "Retrieves every schema belonging to a vendor",
+    notes = "Returns a collection of schemas", httpMethod = "GET", response = classOf[Schema])
   @ApiImplicitParams(Array(
-    new ApiImplicitParam(name = "filter", value = "Metadata filter",
-      required = false, dataType = "string", paramType = "query",
-      allowableValues = "metadata")
+    new ApiImplicitParam(name = "vendor", value = "Comma-separated list of schema vendors",
+      required = true, dataType = "string", paramType = "path", allowMultiple = true),
+    new ApiImplicitParam(name = "filter", value = "Get only schema or only metadata",
+      required = false, dataType = "string", paramType = "query", allowableValues = "metadata"),
+    new ApiImplicitParam(name = "metadata", value = "Include/exclude metadata, choose 1 to include metadata in schemas",
+      required = false, dataType = "string", paramType = "query", allowableValues = "1")
   ))
   @ApiResponses(Array(
-    new ApiResponse(code = 404,
-      message = "There are no public schemas available"),
-    new ApiResponse(code = 401,
-      message = "The supplied authentication is invalid"),
-    new ApiResponse(code = 401, message = """The resource requires
-      authentication, which was not supplied with the request"""),
-    new ApiResponse(code = 404, message = "There are no schemas available here")
+    new ApiResponse(code = 200, message = "{...}"),
+    new ApiResponse(code = 401, message = "The supplied authentication is invalid"),
+    new ApiResponse(code = 401, message = "The resource requires authentication," +
+                                          "which was not supplied with the request"),
+    new ApiResponse(code = 404, message = "There are no schemas for this vendor")
   ))
-  def publicSchemasRoute(owner: String, permission: String) =
-    anyParam('filter.?) { filter =>
+  def readVendorRoute(@ApiParam(hidden = true) v: List[String],
+                      @ApiParam(hidden = true) o: String,
+                      @ApiParam(hidden = true) p: String): Route =
+    (parameter('filter.?) | formField('filter.?)) { filter =>
       filter match {
         case Some("metadata") => complete {
-          (schemaActor ? GetPublicMetadata(owner, permission))
-            .mapTo[(StatusCode, String)]
-        }
-        case _ => complete {
-          (schemaActor ? GetPublicSchemas(owner, permission))
+          (schemaActor ? GetMetadataFromVendor(v, o, p))
             .mapTo[(StatusCode, String)]
         }
+        case _ =>
+          (parameter('metadata.?) | formField('metadata.?)) { metadata =>
+            metadata match {
+              case Some("1") => complete {
```

Comment:
Don't you think we should throw some exception on unexpected values, like `2`?

Reply:
I do, handled by throwing `IllegalArgumentException`.